garage/tf/q_functions/base2.py

```diff
@@ -0,0 +1,44 @@
+"""Q-function base classes without Parameterized."""
+
+
+class QFunction2:
+    """
+    Q-function base class without Parameterzied.
+
+    Args:
+        name (str): Name of the Q-fucntion, also the variable scope.
+
+    """
+
+    def __init__(self, name):
+        self.name = name or type(self).__name__
+        self._variable_scope = None
+
+    def get_qval_sym(self, *input_phs):
+        """
+        Symbolic graph for q-network.
+
+        All derived classes should implement this function.
+
+        Args:
+            input_phs (list[tf.Tensor]): Recommended to be positional
+            arguments, e.g. def get_qval_sym(self, state_input, action_input).
+        """
+        raise NotImplementedError
+
+    def clone(self, name):
+        """
+        Return a clone of the Q-function.
```

Comment:
does this share parameters of the original, or no?

Reply:
It doesn't, only copys the configuration. It is usually used for the target network/q function so we don't actually need to also copy the parameters. I will add that in the docstring.

Reply:
yeah -- just make sure that the behavior is clear